lib/api/route/streams.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:json_annotation/json_annotation.dart';
+
+import '../core.dart';
+part 'streams.g.dart';
+
+/// https://zulip.com/api/get-stream-topics
+Future<GetTopicsResult> getStreamTopics(
+  ApiConnection connection, {
+  required int streamId,
+}) {
+  return connection.get('getStreamTopics', GetTopicsResult.fromJson,
+    'users/me/$streamId/topics', {});
+}
+
+@JsonSerializable(fieldRename: FieldRename.snake)
+class GetTopicsResult {
+  final List<Topic> topics;
+
+  GetTopicsResult({
+    required this.topics,
+  });
+
+  factory GetTopicsResult.fromJson(Map<String, dynamic> json) =>
+    _$GetTopicsResultFromJson(json);
+
+  Map<String, dynamic> toJson() => _$GetTopicsResultToJson(this);
+}
+
+@JsonSerializable(fieldRename: FieldRename.snake)
+class Topic {
+  final int maxId;
+  @JsonKey(name: 'name')
+  final String value;
```

Comment:
`name` is a perfectly reasonable name for this — so let's stay aligned with the server API.